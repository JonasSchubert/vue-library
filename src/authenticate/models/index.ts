export * from './authenticate-module-setup.d';
export * from './authenticate-state.d';
export * from './base-authenticate-state.d';
export * from './login-data.d';
export * from './login-response.d';
export * from './validate-token-data.d';
