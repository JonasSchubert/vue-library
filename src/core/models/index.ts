export * from './base-state.d';
export * from './entity.d';
