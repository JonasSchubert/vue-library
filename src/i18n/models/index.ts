export * from './action-load-payload.d';
export * from './i18n-module-config.d';
export * from './i18n-state.d';
