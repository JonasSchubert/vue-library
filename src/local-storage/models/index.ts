export * from './local-storage-history.d';
export * from './local-storage-state.d';
export * from './local-storage-settings.d';
