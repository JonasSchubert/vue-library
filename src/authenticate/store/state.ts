import { AuthenticateState } from '../models';

export const createAuthenticateState = (): AuthenticateState => ({
  error: undefined,
  isLoading: false,
  token: undefined,
  userRoles: []
});
