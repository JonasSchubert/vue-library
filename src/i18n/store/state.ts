import { I18nModuleConfig, I18nState } from '../models';

export const createState = ({ defaultLocale, defaultMessages }: I18nModuleConfig): I18nState => ({
  availableLocales: [defaultLocale],
  currentLocale: defaultLocale,
  error: undefined,
  locales: { [defaultLocale]: defaultMessages }
});
