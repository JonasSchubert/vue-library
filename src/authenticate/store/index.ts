import { Module } from 'vuex';
import { AuthenticateModuleSetup, AuthenticateState, LoginResponse } from '../models';
import { createActions } from './actions';
import { createGetters } from './getters';
import { createMutations } from './mutations';
import { createAuthenticateState } from './state';

export const createAuthenticateModule = <TRootState, T extends LoginResponse>({ cookieKeyAuthenticationToken, customGetters, daysTilExpiredAuthenticationCookie }: AuthenticateModuleSetup<TRootState, T>): Module<AuthenticateState<T>, TRootState> => ({
  namespaced: true,
  actions: createActions(),
  getters: createGetters(customGetters),
  mutations: createMutations({ cookieKeyAuthenticationToken, daysTilExpiredAuthenticationCookie }),
  state: createAuthenticateState()
});
