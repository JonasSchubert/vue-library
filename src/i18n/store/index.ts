import { Module } from 'vuex';
import { I18nModuleConfig, I18nState } from '../models';
import { createActions } from './actions';
import { createGetters } from './getters';
import { createMutations } from './mutations';
import { createState } from './state';

export const createModule = <T>(config: I18nModuleConfig): Module<I18nState, T> => ({
  namespaced: true,
  actions: createActions<T>(config),
  getters: createGetters<T>(),
  mutations: createMutations(config),
  state: createState(config)
});
