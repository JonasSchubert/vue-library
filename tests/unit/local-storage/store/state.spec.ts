import { LocalStorageState } from '@/local-storage/models';
import { createLocalStorageState } from '@/local-storage/store/state';

describe('LocalStorageState', () => {
  test('should not change unexpected with isDarkTheme true', () => {
    // Act
    const actual: LocalStorageState = createLocalStorageState(true);

    // Assert
    expect(actual).toMatchSnapshot();
  });
  test('should not change unexpected with isDarkTheme false', () => {
    // Act
    const actual: LocalStorageState = createLocalStorageState(false);

    // Assert
    expect(actual).toMatchSnapshot();
  });
});
