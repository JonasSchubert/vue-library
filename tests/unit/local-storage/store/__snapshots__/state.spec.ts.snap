// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocalStorageState should not change unexpected with isDarkTheme false 1`] = `
Object {
  "history": Object {
    "lastRoutes": Array [],
  },
  "settings": Object {
    "isDarkTheme": false,
  },
}
`;

exports[`LocalStorageState should not change unexpected with isDarkTheme true 1`] = `
Object {
  "history": Object {
    "lastRoutes": Array [],
  },
  "settings": Object {
    "isDarkTheme": true,
  },
}
`;
